File: tanklaboratory/src/main/java/com/skx/tomike/tanklaboratory/widget/adapter/RecyclerViewBannerAdapter.java
package com.skx.tomike.tanklaboratory.widget.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.recyclerview.widget.RecyclerView;

import com.skx.tomike.tanklaboratory.R;
import com.skx.tomikecommonlibrary.imageloader.ImageLoader;

import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by shiguotao on 2017/10/13.
 */
public class RecyclerViewBannerAdapter extends RecyclerView.Adapter<RecyclerViewBannerAdapter.ItemViewHolder> {

    private final List<String> mBannerList = new ArrayList<>();

    public RecyclerViewBannerAdapter(List<String> contentList) {
        if (contentList != null) {
            mBannerList.addAll(contentList);
        }
    }

    @Override
    public int getItemCount() {
        return mBannerList.size();
    }

    @NotNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NotNull ViewGroup parent, int viewType) {
        return new ItemViewHolder(LayoutInflater.from(parent.getContext())
                .inflate(R.layout.adapter_recycler_view_banner, parent, false));
    }

    @Override
    public void onBindViewHolder(@NotNull ItemViewHolder holder, int position) {
        ImageLoader.with(holder.itemView.getContext()).load(mBannerList.get(position)).into(holder.mIvImage);
    }

    static class ItemViewHolder extends RecyclerView.ViewHolder {

        ImageView mIvImage;

        ItemViewHolder(View itemView) {
            super(itemView);
            mIvImage = itemView.findViewById(R.id.iv_recyclerViewAsViewPager_bannerImage);
        }
    }
}

File: tank/src/main/java/com/skx/tomike/tank/widget/adapter/RecyclerViewBannerAdapter.kt
package com.skx.tomike.tank.widget.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import androidx.recyclerview.widget.RecyclerView
import com.skx.common.imageloader.ImageLoader
import com.skx.tomike.tank.R
import java.util.ArrayList

class RecyclerViewBannerAdapter(contentList: List<String>?) : RecyclerView.Adapter<RecyclerViewBannerAdapter.ItemViewHolder>() {

    private val mBannerList: MutableList<String> = ArrayList()

    override fun getItemCount(): Int {
        return mBannerList.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        return ItemViewHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.adapter_recycler_view_banner, parent, false))
    }

    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
        ImageLoader.with(holder.itemView.context).load(mBannerList[position]).into(holder.mIvImage)
    }

    class ItemViewHolder internal constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var mIvImage: ImageView = itemView.findViewById(R.id.iv_recyclerViewAsViewPager_bannerImage)

        init {
            val layoutParams = mIvImage.layoutParams as RelativeLayout.LayoutParams
            layoutParams.topMargin = 15
            layoutParams.bottomMargin = 15
            layoutParams.leftMargin = 15
            layoutParams.rightMargin = 15
            mIvImage.layoutParams = layoutParams
        }
    }

    init {
        if (contentList != null) {
            mBannerList.addAll(contentList)
        }
    }
}